src/cli/self_update.rs

```diff
@@ -1128,6 +1125,10 @@ pub(crate) fn update(cfg: &Cfg) -> Result<utils::ExitCode> {
     Ok(utils::ExitCode(0))
 }
 
+fn get_and_parse_new_rustup_version(path: &Path) -> Option<String> {
```

Comment:
I'm not sure this one-off function carries its own weight.

```
  let Some(version) = get_new_rustup_version(&setup_path).map(parse_new_rustup_version) else {...}
```

doesn't have enough complexity, reusability, or cyclomatic complexity to worry about

Reply:
nvm I see its reused